test/components/legendTests.ts

```diff
@@ -254,6 +254,34 @@ describe("Legend", () => {
     svg.remove();
   });
 
+  it("accept formatter and show entity labels correcly", () => {
+    color.domain(["jtao", "mschafer", "kfalter"]);
+    let data = [
+      {id: "jtao", name: "Joy Tao", score: 13},
+      {id: "mschafer", name: "Mark Schafer", score: 15},
+      {id: "kfalter", name: "Kelsey Falter", score: 11}
+    ];
+    let f = (id: string) => {
+      let pickedEntity = data.filter(function(d) {return (d.id === id); });
+      if (pickedEntity.length > 0) {
+          return pickedEntity[0].name;
+        }else {
+          return "NA";
+        }
```

Comment:
Instead of the logic above, why can't we create a map instead of a data array?  It'd be immensely easier to work with here.

Just noting that it seems you're adding logic to this formatter that seems unnecessary (this "NA" case, the `pickedEntity.length > 0` case), which can make deciphering this test harder to understand when we look at this later.